docs/src/pages/components/data-grid/getting-started/migration-v4.md

```diff
@@ -0,0 +1,433 @@
+  | `.MuiGridFilterForm-root`          | `.MuiDataGrid-filterForm`                |
+  | `.MuiGridToolbarFilterButton-root` | `.MuiDataGrid-toolbarFilterList`         |
+
+### Module augmentation
+
+- The module augmentation is no longer enabled by default. This change was done to prevent conflicts with projects using `DataGrid` and `DataGridPro` together.
+
+  In order to still be able to do overrides at the theme level, add the following imports to your project:
+
+  ```diff
+  +import type {} from '@mui/x-data-grid/themeAugmentation';
+  +import type {} from '@mui/x-data-grid-pro/themeAugmentation';
+  ```
+
+### Virtualization
+
+- The `onViewportRowsChange` prop and the `viewportRowsChange` event have been removed
+
+  You can replace them with a listener on the `rowsScroll` event:
+
+  ```tsx
+  const apiRef = useGridApiRef();
+  const prevRenderContext = React.useRef(null);
+
+  React.useEffect(() => {
+    return apiRef.current.subscribeEvent('rowsScroll', ({ renderContext }) => {
+      if (
+        !prevRenderContext.current ||
+        renderContext.firstRowIdx !== prevRenderContext.current.firstRowIndex ||
+        renderContext.lastRowIdx !== prevRenderContext.current.lastRowIndex
+      ) {
+        prevRenderContext.current = renderContext;
+        const params = {
+          firstRowIndex: renderContext.firstRowIndex,
+          lastRowIndex: renderContext.lastRowIndex,
+        };
+      }
+    });
+  }, [apiRef]);
+
+  <DataGridPro apiRef={apiRef} />;
+  ```
+
+### Properties of the Grid
+
+- Some event listeners and DOM attributes were removed from `GridCell` and `GridRow` to improve performances.
+
+  The following props were removed.
+
+  - `onCellBlur`
+  - `onCellOver`
+  - `onCellOut`
+  - `onCellEnter`
+  - `onCellLeave`
+  - `onRowOver`
+  - `onRowOut`
+  - `onRowEnter`
+  - `onRowLeave`
+
+  If you depend on them, you can use `componentsProps.row` and `componentsProps.cell` to pass custom props to the row or cell.
+  For more information, check [this page](https://mui.com/components/data-grid/components/#row). Example:
+
+  ```diff
+  -<DataGrid onRowOver={handleRowOver} />;
+  +<DataGrid
+  +  componentsProps={{
+  +    row: { onMouseOver: handleRowOver },
+  +  }}
+  +/>;
+  ```
+
+  The `data-rowindex` and `data-rowselected` attributes were removed from the cell element. Equivalent attributes can be found in the row element.
+
+  The `data-editable` attribute was removed from the cell element. Use the `.MuiDataGrid-cell--editable` CSS class.
+
+  The `data-mode` attribute was removed from the cell element. Use the `.MuiDataGrid-cell--editing` CSS class.
+
+### State access
+
+- The state direct access is not considered part of the public API anymore. We only guarantee that the selectors continue to work between minor releases.
+  We advise you to avoid accessing directly a state sub-key and to use selectors or `apiRef` methods whenever possible.
```

Comment:
```suggestion
  We advise you to avoid accessing directly a state sub-key, instead use the available selectors or `apiRef` methods whenever possible.
```